### ISSUE TITLE
Modify time of keyframe

### ISSUE THREAD

**Issue by byumark** [airbnb/lottie-web#1872]
Awesome tool! 

Is there a way to shift the time of keyframes dynamically? Inside of after effects I was able to accomplish this with something like `valueAtTime(time-2)`, but is there a way to make that `2` dynamic so that we can provide that value while loading the animation?

Would I have to overwrite a property with something like `var $bm_rt;\n$bm_rt = valueAtTime($bm_sub(time, 2));`?

**Comment by Inventsable** [comment]
Hello, maybe reference the 2 as the value of a slider control within AE, defender with the new expression, and then use lottie-api.js to change the value of this layer.effect on launch?

**Comment by byumark** [comment]
Good thinking. I'll give that a try. Thanks!

**Comment by byumark** [comment]
I'm trying to implement this following the example file:

**Comment by byumark** [comment]
I'm trying to implement this following the example file:
```
var mousePosition = [0,0]
animationAPI = lottie_api.createAnimationApi(anim);
var positionProperty = animationAPI.getKeyPath('red_solid,Transform,Position');
animationAPI.addValueCallback(positionProperty, function(currentValue) {
    return mousePosition;
});
```
I'm not understanding how this is working. Is the property value supposed to change whenever the `mousePosition` value changes? Also, how does the `'red_solid,Transform,Position'` change if the target layer is inside of a preComp? I can't seem to target my property.

**Comment by bodymovin** [comment]
Yes, if you update the mousePosition value it will be picked up by the method you are passing to addValueCallback.
Regarding precomps you should add the precomp layer name as part of the path. Something like:
'Precomp Name,red_solid,Transform,Position'

**Comment by byumark** [comment]
Great, thank you. How do you target a text layer's `Source Text` property to change the text? I can't seem to get it to work.

**Comment by Inventsable** [comment]
Not sure if this is still up to date:

https://github.com/airbnb/lottie-web/wiki/TextLayer.updateDocumentData

There may have been a more recent Issue filing that solved this with an updated solution, I'd search the Issues here too

**Comment by kudanai** [comment]
Hi there. I'm closing this issue for the time being, since it's been stale for a while and there have been a significant number of updates. Please feel free to re-open it or file a new issue.
